`TestConnectionWrapper_Read` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/4493152265/jobs/7903987200?pr=23416
- https://github.com/gravitational/teleport/actions/runs/4493152265/jobs/7907650549?pr=23416
- https://github.com/gravitational/teleport/actions/runs/4608626090/jobs/8144653050?pr=23865

#### Relevant snippet

```
=== RUN   TestConnectionWrapper_Read
=== RUN   TestConnectionWrapper_Read/Plain_connection_without_any_special_headers
--- PASS: TestConnectionWrapper_Read/Plain_connection_without_any_special_headers (0.01s)
=== RUN   TestConnectionWrapper_Read/Sending_ProxyHelloSignature
--- PASS: TestConnectionWrapper_Read/Sending_ProxyHelloSignature (0.02s)
=== RUN   TestConnectionWrapper_Read/Sending_PROXY_header
panic: Fail in goroutine after TestConnectionWrapper_Read/Sending_ProxyHelloSignature has completed

goroutine 10 [running]:
testing.(*common).Fail(0xc000594000)
	/opt/go/src/testing/testing.go:933 +0x1a8
testing.(*common).Errorf(0xc000594000, {0x1b5fb6f, 0x3}, {0xc0000da140, 0x1, 0x1})
	/opt/go/src/testing/testing.go:1050 +0xaf
github.com/stretchr/testify/assert.Fail({0x1d76600, 0xc000594000}, {0xc0002de070, 0x66}, {0x0, 0x0, 0x0})
	/go/pkg/mod/github.com/stretchr/testify@v1.8.2/assert/assertions.go:262 +0x412
github.com/stretchr/testify/assert.NoError({0x1d76600, 0xc000594000}, {0x1d761c0, 0xc0001144b0}, {0x0, 0x0, 0x0})
	/go/pkg/mod/github.com/stretchr/testify@v1.8.2/assert/assertions.go:1380 +0x127
github.com/gravitational/teleport/lib/sshutils.startSSHServer.func3()
	/__w/teleport/teleport/lib/sshutils/server_test.go:374 +0x6c
created by github.com/gravitational/teleport/lib/sshutils.startSSHServer
	/__w/teleport/teleport/lib/sshutils/server_test.go:368 +0x7da
FAIL	github.com/gravitational/teleport/lib/sshutils	0.115s
```

--- a/lib/multiplexer/multiplexer_test.go
+++ b/lib/multiplexer/multiplexer_test.go
@@ -1113,7 +1113,9 @@ func startSSHServer(t *testing.T, listener net.Listener) {
 	go func() {
 		for newReq := range reqs {
 			if newReq.Type == "echo" {
-				newReq.Reply(true, newReq.Payload)
+				err := newReq.Reply(true, newReq.Payload)
+				assert.NoError(t, err)
+				continue
 			}
 			err := newReq.Reply(false, nil)
 			assert.NoError(t, err)

--- a/lib/sshutils/server_test.go
+++ b/lib/sshutils/server_test.go
@@ -368,7 +368,9 @@ func startSSHServer(t *testing.T, listener net.Listener) {
 	go func() {
 		for newReq := range reqs {
 			if newReq.Type == "echo" {
-				newReq.Reply(true, newReq.Payload)
+				err := newReq.Reply(true, newReq.Payload)
+				assert.NoError(t, err)
+				continue
 			}
 			err := newReq.Reply(false, nil)
 			assert.NoError(t, err)

